### DISCUSSION TITLE
uniqueness of the task podname [k8s-platform]

### DISCUSSION THREAD

**Discussion by naineshmp** [nextflow-io/nextflow#3467] [Q&A]
I am exploring the config k8s.cleanup which skips podCleanup after successful execution of the task. 

It seems `pod.name` is set as `task.hash` and `task.hash` seems to depend on inputs, session.id, def, name etc. 

Question - If I have two pipeline steps, with similar definition and inputs, and is there a clash for pod-name? If I run these two tasks in serial, only one of them will be successful? Sorry, if this is already explained somewhere. I am just curious about uniqueness of task-pod-name.

Currently, pod_name (hash) seems to be using multiple keys [ session.uniqueId, name, task.source ] + inputs 
https://github.com/nextflow-io/nextflow/blob/41021cbcbf3f2652d263ec9b7a6ed311ca1a47eb/modules/nextflow/src/main/groovy/nextflow/processor/TaskProcessor.groovy#L2015-L2022

**Comment by pditommaso** [comment]
Each task is associated with a hash identity computed using the task inputs, the command to be executed and the pipeline execution unique identifier (session-id). Therefore two identical tasks in different pipelines will get a different hash id -> pod name
